### ISSUE TITLE
Versioning changes

### ISSUE THREAD

**Issue by morrone** [ovis-hpc/ovis#95]
In #94, I mentioned that it would probably be good to adopt Semantic Versioning for ovis/lmds.

https://semver.org/

Next, I would like to request that official tags be "annotated" tags from now on.  This just requires doing "git tag -a".  Annotated tags allow comments, but more importantly they actually record who made the tag and when.  If you want to get fancy, you can use signed tags "git tag -s", which are annotated tags that are gpg signed.

Also, it might be good to revamp the way the version number is handled in the build system.  Ovis has some parts to record git version information, as seen in SHA.txt, TAG.txt (in the top level Makefile.am), and to some extent in automake/pedigree.am.

I am going to suggest that we can replace some of that with a script in AC_INIT to generate the version from git describe information (which means it can detect when a commit is tagged, and use the tagged version).  This would normally cause issues when using the dist tarball, because git is not available at that time.  But the clever git people found a reasonable way to do that (since git is an autotools built project, and is itself housed in a git repo).

The git project's solution was the GIT-VERSION-GEN script, and I have reworked that for other projects in the past.  But it looks like the autoconf project came up with a version that is meant to be more resuable than git's original script.  The script can be found here in autoconf's source repo:

http://git.savannah.gnu.org/gitweb/?p=autoconf.git;a=blob;f=build-aux/git-version-gen;hb=HEAD

The comments in the script explain how it works, and how to deploy it in a project. (This script, as written, requires using annotated tags).

The nice thing we would get from doing this is that the built ldms version would use the information from git-describe.  

For instance, right now I have a branch that is one commit ahead of the OVIS-4.3.3-beta2 tag.  The git describe output gives me:

```
OVIS-4.3.3-beta2-1-g0969b10
```

The method above would turn that into a version string that looks like:

```
4.3.3-beta2.1-0969
```

If there is an edited file in my working directory, I would get this:

```
4.3.3-beta2.1-0969-dirty
```

And, of course, if my checkout commit is the same as tag OVIS-4.3.3-beta2, and my working directory is clean, I would get:

```
4.3.3-beta2
```

And a nice minor side effect is that the release manager no longer needs to create a "bumping the version number" commit any more.

**Comment by morrone** [comment]
@oceandlr, @valleydlr, @tom95858, @narategithub: I'm not sure who will make the 4.3.3 tag when it is time.  Whoever it is, could you please use an annotated tag for that (just use "git tag -a").  An annotated tag has an associated message that you can use (or not) as you wish, and also records who made the tag and when (just like a commit).  But more importantly, commands like "git describe" expect that the important version tags will be annotated and ignore (by default) lightweight tags.  They work that way because annotated tags are generally standard practice now for published tags.

**Comment by morrone** [comment]
This is largely addressed by the landing of PR #267 on the OVIS-4 branch.
